ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpStatement.kt

```diff
@@ -39,8 +43,14 @@ public class HttpStatement(
      *
      * Please note: the [response] instance will be canceled and shouldn't be passed outside of [block].
      */
-    public suspend fun <T> execute(block: suspend (response: HttpResponse) -> T): T {
-        val response: HttpResponse = executeUnsafe()
+    public suspend fun <T> execute(
+        listener: ProgressListener? = null,
```

Comment:
I'm not sure that we should introduce such an API call for the specific feature. Could we make an extension point in the request builder instead?